include/picongpu/particles/collision/detail/ListEntry.hpp

```diff
@@ -28,273 +28,386 @@
-                DINLINE auto getParticle(T_ParBox& parBox, T_FramePtr frame, uint32_t particleId) ->
-                    typename T_FramePtr::type::ParticleType
-                {
-                    constexpr int frameSize = T_ParBox::frameSize;
-                    uint32_t const skipFrames = particleId / frameSize;
-                    for(uint32_t i = 0; i < skipFrames; ++i)
-                        frame = parBox.getNextFrame(frame);
-                    return frame[particleId % frameSize];
-                }
+        /* Swap two list entries.
+         *
+         * @param v0 index of the 1st entry to swap
+         * @param v0 index of the 2nd entry to swap
+         */
+        DINLINE void swap(uint32_t& v0, uint32_t& v1)
+        {
+            uint32_t tmp = v0;
+            v0 = v1;
+            v1 = tmp;
+        }
 
-                template<
-                    typename T_Worker,
-                    typename T_ForEach,
-                    typename T_DeviceHeapHandle,
-                    typename T_ParBox,
-                    typename T_FramePtr,
-                    typename T_EntryListArray,
-                    typename T_Array,
-                    typename T_Filter>
-                DINLINE void prepareList(
-                    T_Worker const& worker,
-                    T_ForEach forEach,
-                    T_DeviceHeapHandle deviceHeapHandle,
-                    T_ParBox& parBox,
-                    T_FramePtr firstFrame,
-                    uint32_t const numParticlesInSupercell,
-                    T_EntryListArray& parCellList,
-                    T_Array& nppc,
-                    T_Filter filter)
-                {
-                    // Initialize nppc with zeros.
-                    forEach([&](uint32_t const linearIdx) { nppc[linearIdx] = 0u; });
-                    worker.sync();
-                    // Count eligible
-                    particlesCntHistogram(worker, forEach, parBox, firstFrame, numParticlesInSupercell, nppc, filter);
-                    worker.sync();
-
-                    // memory for particle indices
-                    forEach([&](uint32_t const linearIdx)
-                            { parCellList[linearIdx].init(worker, deviceHeapHandle, nppc[linearIdx]); });
-                    worker.sync();
-
-                    detail::updateLinkedList(
-                        worker,
-                        forEach,
-                        parBox,
-                        firstFrame,
-                        numParticlesInSupercell,
-                        parCellList,
-                        filter);
-                    worker.sync();
-                }
-            } // namespace detail
-        } // namespace collision
-    } // namespace particles
-} // namespace picongpu
+        /* Shuffle list entries
+         *
+         * Shuffle the given array with indicies.
+         *
+         * @param worker lockstep worker
```

Comment:
twice 'workeer'?

Reply:
1. is the parameter name the other is part of the description